app/models/tag.rb

```diff
@@ -0,0 +1,11 @@
+class Tag < ApplicationRecord
+  validates_presence_of :title
+  validates_uniqueness_of :title, case_sensitive: false
+
+  has_many :post_tags, dependent: :destroy
+  has_many :posts, through: :post_tags
+
+  scope :order_by_most_posts, -> { unscoped.order(posts_count: :desc) }
```

Comment:
I removed a similar scope from the user model, I separated this behaviour to a service to make it work to multiple models, that makes it cleaner from my point of view